src/Symfony/Component/Mailer/Transport/Smtp/Stream/SocketStream.php

```diff
@@ -167,7 +167,14 @@ public function initialize(): void
 
     public function startTLS(): bool
     {
-        return (bool) stream_socket_enable_crypto($this->stream, true);
+        set_error_handler(function ($type, $msg) {
+            throw new TransportException('Unable to connect with STARTTLS: '.$msg);
+        });
+        try {
+            return stream_socket_enable_crypto($this->stream, true);
```

Comment:
you remove the `(bool)` cast. Is it expected ?

Reply:
the function returns `int` only when a non blocking stream is passes, which is never the case here and would be unsupported anyway